File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
' ...

Namespace RunningSum
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private currentTotal As Integer = 0

		Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
			currentTotal += Convert.ToInt32(Me.GetCurrentColumnValue("UnitsOnOrder"))
		End Sub

		Private Sub GroupFooter1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles GroupFooter1.BeforePrint
			Me.xrLabel3.Text = currentTotal.ToString()
		End Sub

	End Class
End Namespace

File: CS/XtraReport1.cs
using System;
// ...

namespace RunningSum {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private int currentTotal = 0;

        private void Detail_BeforePrint(object sender, 
        System.Drawing.Printing.PrintEventArgs e) {
            currentTotal += Convert.ToInt32(this.GetCurrentColumnValue("UnitsOnOrder"));
        }

        private void GroupFooter1_BeforePrint(object sender, 
        System.Drawing.Printing.PrintEventArgs e) {
            this.xrLabel3.Text = currentTotal.ToString();
        }

    }
}
